llm_bench/python/benchmark.py

```diff
@@ -145,11 +147,15 @@ def run_text_generation(input_text, num, model, tokenizer, args, iter_data_list,
 
 
 def run_text_generation_benchmark(model_path, framework, device, args, num_iters):
-    bench_hook = HOOK_UTILS[framework].BenchHook()
     model, tokenizer, pretrain_time = FW_UTILS[framework].create_text_gen_model(model_path, device, **args)
     # Override forward for statistic each forward time.
     default_model_type = DEFAULT_MODEL_CLASSES[args['use_case']]
     model_type = args.get('model_type', default_model_type)
+
+    if args['num_beams'] > 1:
```

Comment:
Should be >0? @xipingyan

Reply:
As I known, `num_beams` should be greater than 0, 1=`gready_search`.

Reply:
hi @wgzintel @peterchen-intel , please see internal logic:
`elif not generation_config.do_sample and generation_config.num_beams > 1:`
If this condition is met, beam search will be called.

Reply:
@xipingyan Got it.  Whether to select greedy search or beam search from the internal logic of transformers through the _get_generation_mode function
https://github.com/huggingface/transformers/blob/514de24abfd4416aeba6a6455ad5920f57f3567d/src/transformers/generation/utils.py#L858